### ISSUE TITLE
objc-tests failure on Kokoro

### ISSUE THREAD

**Issue by adelez** [grpc/grpc#12149]
https://sponge.corp.google.com/target?id=32e3efb4-5a18-473f-a297-e1a6732c589a&target=github/grpc/objc_macos_dbg_native&searchFor=&show=FAILED&sortBy=STATUS

```
Failure
mulator/gRPC-Core.build/DerivedSources/i386 -I/Users/kbuilder/Library/Developer/Xcode/DerivedData/Tests-giodclofueabqfanadigbvvvjndw/Build/Intermediates/Pods.build/Test-iphonesimulator/gRPC-Core.build/DerivedSources -F/Users/kbuilder/Library/Developer/Xcode/DerivedData/Tests-giodclofueabqfanadigbvvvjndw/Build/Products/Test-iphonesimulator/gRPC-Core -DGRPC_ARES=0 -fno-objc-arc -include /Volumes/BuildData/tmpfs/src/github/grpc/workspace_objc_macos_dbg_native/src/objective-c/tests/Pods/Target\ Support\ Files/gRPC-Core/gRPC-Core-prefix.pch -MMD -MT dependencies -MF /Users/kbuilder/Library/Developer/Xcode/DerivedData/Tests-giodclofueabqfanadigbvvvjndw/Build/Intermediates/Pods.build/Test-iphonesimulator/gRPC-Core.build/Objects-normal/i386/error.d --serialize-diagnostics /Users/kbuilder/Library/Developer/Xcode/DerivedData/Tests-giodclofueabqfanadigbvvvjndw/Build/Intermediates/Pods.build/Test-iphonesimulator/gRPC-Core.build/Objects-normal/i386/error.dia -c /Volumes/BuildData/tmpfs/src/github/grpc/workspace_objc_macos_dbg_native/src/core/lib/iomgr/error.c -o /Users/kbuilder/Library/Developer/Xcode/DerivedData/Tests-giodclofueabqfanadigbvvvjndw/Build/Intermediates/Pods.build/Test-iphonesimulator/gRPC-Core.build/Objects-normal/i386/error.o
CompileC /Users/kbuilder/Library/Developer/Xcode/DerivedData/Tests-giodclofueabqfanadigbvvvjndw/Build/Intermediates/Pods.build/Test-iphonesimulator/gRPC-Core.build/Objects-normal/x86_64/error.o /Volumes/BuildData/tmpfs/src/github/grpc/workspace_objc_macos_dbg_native/src/core/lib/iomgr/error.c normal x86_64 c com.apple.compilers.llvm.clang.1_0.compiler
    /Applications/Xcode_8.2.1.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -x c -arch x86_64 -fmessage-length=0 -fdiagnostics-show-note-include-stack -fmacro-backtrace-limit=0 -std=gnu99 -fmodules -fmodules-cache-path=/Users/kbuilder/Library/Developer/Xcode/DerivedData/ModuleCache -fmodules-prune-interval=86400 -fmodules-prune-after=345600 -fbuild-session-file=/Users/kbuilder/Library/Developer/Xcode/DerivedData/ModuleCache/Session.modulevalidation -fmodules-validate-once-per-build-session -Wnon-modular-include-in-framework-module -Werror=non-modular-include-in-framework-module -Wno-trigraphs -fpascal-strings -Os -fno-common -Werror -Wno-missing-field-initializers -Wno-missing-prototypes -Wno-return-type -Wdocumentation -Wunreachable-code -Wno-missing-braces -Wparentheses -Wswitch -Wunused-function -Wno-unused-label -Wno-unused-parameter -Wunused-variable -Wunused-value -Wempty-body -Wuninitialized -Wno-unknown-pragmas -Wno-shadow -Wno-four-char-constants -Wno-conversion -Wconstant-conversion -Wint-conversion -Wbool-conversion -Wenum-conversion -Wshorten-64-to-32 -Wpointer-sign -Wno-newline-eof -DPOD_CONFIGURATION_TEST=1 -DCOCOAPODS=1 -DCOCOAPODS=1 -DGRPC_CRONET_WITH_PACKET_COALESCING=1 -isysroot /Applications/Xcode_8.2.1.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator10.2.sdk -fasm-blocks -fstrict-aliasing -Wdeprecated-declarations -mios-simulator-version-min=7.0 -g -Wno-sign-conversion -Winfinite-recursion -iquote/Volumes/BuildData/tmpfs/src/github/grpc/workspace_objc_macos_dbg_native/src/objective-c/tests/Pods/../../../.. -I/Users/kbuilder/Library/Developer/Xcode/DerivedData/Tests-giodclofueabqfanadigbvvvjndw/Build/Products/Test-iphonesimulator/gRPC-Core/include -I/Volumes/BuildData/tmpfs/src/github/grpc/workspace_objc_macos_dbg_native/src/objective-c/tests/Pods/Headers/Private -I/Volumes/BuildData/tmpfs/src/github/grpc/workspace_objc_macos_dbg_native/src/objective-c/tests/Pods/Headers/Private/gRPC-Core -I/Volumes/BuildData/tmpfs/src/github/grpc/workspace_objc_macos_dbg_native/src/objective-c/tests/Pods/Headers/Public -I/Volumes/BuildData/tmpfs/src/github/grpc/workspace_objc_macos_dbg_native/src/objective-c/tests/Pods/Headers/Public/!ProtoCompiler -I/Volumes/BuildData/tmpfs/src/github/grpc/workspace_objc_macos_dbg_native/src/objective-c/tests/Pods/Headers/Public/!ProtoCompiler-gRPCPlugin -I/Volumes/BuildData/tmpfs/src/github/grpc/workspace_objc_macos_dbg_native/src/objective-c/tests/Pods/Headers/Public/BoringSSL -I/Volumes/BuildData/tmpfs/src/github/grpc/workspace_objc_macos_dbg_native/src/objective-c/tests/Pods/Headers/Public/CronetFramework -I/Volumes/BuildData/tmpfs/src/github/grpc/workspace_objc_macos_dbg_native/src/objective-c/tests/Pods/Headers/Public/Protobuf -I/Volumes/BuildData/tmpfs/src/github/grpc/workspace_objc_macos_dbg_native/src/objective-c/tests/Pods/Headers/Public/RemoteTest -I/Volumes/BuildData/tmpfs/src/github/grpc/workspace_objc_macos_dbg_native/src/objective-c/tests/Pods/Headers/Public/gRPC -I/Volumes/BuildData/tmpfs/src/github/grpc/workspace_objc_macos_dbg_native/src/objective-c/tests/Pods/Headers/Public/gRPC-Core -I/Volumes/BuildData/tmpfs/src/github/grpc/workspace_objc_macos_dbg_native/src/objective-c/tests/Pods/Headers/Public/gRPC-ProtoRPC -I/Volumes/BuildData/tmpfs/src/github/grpc/workspace_objc_macos_dbg_native/src/objective-c/tests/Pods/Headers/Public/gRPC-RxLibrary -I/Volumes/BuildData/tmpfs/src/github/grpc/workspace_objc_macos_dbg_native/src/objective-c/tests/Pods/Headers/Public/nanopb -I/Volumes/BuildData/tmpfs/src/github/grpc/workspace_objc_macos_dbg_native/src/objective-c/tests/Pods/../../../../include -I/Users/kbuilder/Library/Developer/Xcode/DerivedData/Tests-giodclofueabqfanadigbvvvjndw/Build/Intermediates/Pods.build/Test-iphonesimulator/gRPC-Core.build/DerivedSources/x86_64 -I/Users/kbuilder/Library/Developer/Xcode/DerivedData/Tests-giodclofueabqfanadigbvvvjndw/Build/Intermediates/Pods.build/Test-iphonesimulator/gRPC-Core.build/DerivedSources -F/Users/kbuilder/Library/Developer/Xcode/DerivedData/Tests-giodclofueabqfanadigbvvvjndw/Build/Products/Test-iphonesimulator/gRPC-Core -DGRPC_ARES=0 -fno-objc-arc -include /Volumes/BuildData/tmpfs/src/github/grpc/workspace_objc_macos_dbg_native/src/objective-c/tests/Pods/Target\ Support\ Files/gRPC-Core/gRPC-Core-prefix.pch -MMD -MT dependencies -MF /Users/kbuilder/Library/Developer/Xcode/DerivedData/Tests-giodclofueabqfanadigbvvvjndw/Build/Intermediates/Pods.build/Test-iphonesimulator/gRPC-Core.build/Objects-normal/x86_64/error.d --serialize-diagnostics /Users/kbuilder/Library/Developer/Xcode/DerivedData/Tests-giodclofueabqfanadigbvvvjndw/Build/Intermediates/Pods.build/Test-iphonesimulator/gRPC-Core.build/Objects-normal/x86_64/error.dia -c /Volumes/BuildData/tmpfs/src/github/grpc/workspace_objc_macos_dbg_native/src/core/lib/iomgr/error.c -o /Users/kbuilder/Library/Developer/Xcode/DerivedData/Tests-giodclofueabqfanadigbvvvjndw/Build/Intermediates/Pods.build/Test-iphonesimulator/gRPC-Core.build/Objects-normal/x86_64/error.o
CpHeader /Volumes/BuildData/tmpfs/src/github/grpc/workspace_objc_macos_dbg_native/src/core/lib/iomgr/error.h /Users/kbuilder/Library/Developer/Xcode/DerivedData/Tests-giodclofueabqfanadigbvvvjndw/Build/Products/Test-iphonesimulator/gRPC-Core/error.h
    builtin-copy -exclude .DS_Store -exclude CVS -exclude .svn -exclude .git -exclude .hg -strip-debug-symbols -strip-tool /Applications/Xcode_8.2.1.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/strip -resolve-src-symlinks /Volumes/BuildData/tmpfs/src/github/grpc/workspace_objc_macos_dbg_native/src/core/lib/iomgr/error.h /Users/kbuilder/Library/Developer/Xcode/DerivedData/Tests-giodclofueabqfanadigbvvvjndw/Build/Products/Test-iphonesimulator/gRPC-Core
=== BUILD TARGET gRPC OF PROJECT Pods WITH CONFIGURATION Test ===
=== BUILD TARGET gRPC-ProtoRPC OF PROJECT Pods WITH CONFIGURATION Test ===
=== BUILD TARGET RemoteTest OF PROJECT Pods WITH CONFIGURATION Test ===
=== BUILD TARGET Pods-Tests OF PROJECT Pods WITH CONFIGURATION Test ===
=== BUILD TARGET Pods-AllTests OF PROJECT Pods WITH CONFIGURATION Test ===
=== BUILD TARGET Tests OF PROJECT Tests WITH CONFIGURATION Test ===
=== BUILD TARGET AllTests OF PROJECT Tests WITH CONFIGURATION Test ===
ld: warning: directory not found for option '-F/Applications/Xcode_8.2.1.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator10.2.sdk/Developer/Library/Frameworks'
kill -9 `jobs -p` ; echo "EXIT TIME:  $(date)"
jobs -p
date
EXIT TIME:  Thu Aug 10 09:14:20 PDT 2017
```

**Comment by dgquintas** [comment]
This is [still happening](https://sponge.corp.google.com/target?id=60437782-b0d7-412e-8f72-e18ade4432d1&target=grpc/macos/master/grpc_basictests_opt&searchFor=&show=FAILED&sortBy=STATUS) for about 16% of runs. 

@muxi any idea what the cause is and how to tackle it?

**Comment by muxi** [comment]
Actual error message (from https://sponge.corp.google.com/target?id=1cdcea52-649c-4dd2-93eb-c07410c53627&target=github/grpc/objc_macos_opt_native&searchFor=&show=FAILED&sortBy=STATUS)

```
E1206 13:55:46.636612000 123145308282880 cronet_transport.cc:404]      on_failed(0x7ffdfd108d50, -312)
E1206 13:55:46.636852000 4377731072 cq_verifier.cc:273]                cq returned unexpected event: OP_COMPLETE: tag:0x1 OK
E1206 13:55:46.637027000 4377731072 cq_verifier.cc:280]                expected tags:
tag(101) GRPC_OP_COMPLETE success=1 /Volumes/BuildData/tmpfs/src/github/grpc/workspace_objc_macos_opt_native/test/core/end2end/tests/simple_request.cc:163
Restarting after unexpected exit or crash in CoreCronetEnd2EndTests/testSimpleRequest; summary will include totals from previous launches.
```

**Comment by muxi** [comment]
Fixed by #13708
